src/vanilla/store.ts

```diff
@@ -227,19 +226,27 @@ const flushPending = (pending: Pending) => {
 
 // internal & unstable type
 type StoreArgs = readonly [
-  getAtomState: <Value>(atom: Atom<Value>) => AtomState<Value>,
+  getAtomState: <Value>(
+    pending: Pending | undefined,
+    atom: Atom<Value>,
+  ) => AtomState<Value>,
   atomRead: <Value>(
+    pending: Pending | undefined,
     atom: Atom<Value>,
     ...params: Parameters<Atom<Value>['read']>
   ) => Value,
   atomWrite: <Value, Args extends unknown[], Result>(
+    pending: Pending,
     atom: WritableAtom<Value, Args, Result>,
     ...params: Parameters<WritableAtom<Value, Args, Result>['write']>
   ) => Result,
   atomOnMount: <Value, Args extends unknown[], Result>(
+    pending: Pending,
     atom: WritableAtom<Value, Args, Result>,
     setAtom: (...args: Args) => Result,
   ) => OnUnmount | void,
+  createPending: (prevPending?: Pending | undefined) => Pending,
```

Comment:
prevPending is the in-context upstream pending. For instance, unsubscribe's invocation of createPending receives subscribe's pending.